rust/arrow/src/buffer.rs

```diff
@@ -194,7 +195,7 @@ impl Buffer {
 
     /// Returns the byte slice stored in this buffer
     pub fn data(&self) -> &[u8] {
-        unsafe { std::slice::from_raw_parts(self.raw_data(), self.len()) }
+        &self.data.data()[self.offset..]
     }
 
     /// Returns a slice of this buffer, starting from `offset`.
```

Comment:
Isn't the offset assertion in `fn slice` wrong? I was going to talk about overflow/wraparound problems (logical, can't prove a soundness issue), but `self.len()` already subtracts the self.offset once.